aptos-move/aptos-vm/src/sharded_block_executor/sharded_aggregator_service.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_state_view::StateView;
+use aptos_types::{
+    state_store::state_key::StateKey, transaction::TransactionOutput,
+    write_set::TOTAL_SUPPLY_STATE_KEY,
+};
+use rayon::prelude::*;
+use serde::de::DeserializeOwned;
+use std::ops;
+
+pub fn get_state_value<S: StateView, T: DeserializeOwned>(
+    state_key: &StateKey,
+    state_view: &S,
+) -> Option<T> {
+    let value = state_view
+        .get_state_value_bytes(state_key)
+        .ok()?
+        .map(move |value| bcs::from_bytes(value.as_slice()));
+    value.transpose().map_err(anyhow::Error::msg).unwrap()
+}
+
+/// This class ensures that deltas can use all 128 bits without having to let go of the sign bit for
+/// cases where the delta is negative. That is, we don't have to use conversions to i128.
+/// However, it does not handle overflow and underflow. That is, it will indicate to the caller of
+/// the faulty logic with their usage of deltas.
+#[derive(Clone, Copy)]
+struct DeltaU128 {
+    delta: u128,
+    is_positive: bool,
+}
+
+impl DeltaU128 {
+    pub fn get_delta(minuend: u128, subtrahend: u128) -> Self {
+        if minuend >= subtrahend {
+            Self {
+                delta: minuend - subtrahend,
+                is_positive: true,
+            }
+        } else {
+            Self {
+                delta: subtrahend - minuend,
+                is_positive: false,
+            }
+        }
+    }
+
+    fn add_delta(self, other: u128) -> u128 {
+        if self.is_positive {
+            self.delta + other
+        } else {
+            other - self.delta
+        }
+    }
+}
+
+impl Default for DeltaU128 {
+    fn default() -> Self {
+        Self {
+            delta: 0,
+            is_positive: true,
+        }
+    }
+}
+
+impl ops::Add for DeltaU128 {
+    type Output = Self;
+
+    fn add(self, rhs: Self) -> Self::Output {
+        if self.is_positive == rhs.is_positive {
+            Self {
```

Comment:
This function's readability can be improved by reducing if nesting levels..how about returning `Self` from here and remove the else condition below?

Reply:
done and added comments